psp/models/recent_history.py

```diff
@@ -125,19 +125,24 @@ def __init__(
         *,
         pv_data_source: PvDataSource,
         nwp_data_source: NwpDataSource | None,
+        excarta_data_source: NwpDataSource | None,
         regressor: Regressor,
         random_state: np.random.RandomState | None = None,
         use_nwp: bool = True,
+        use_excarta: bool = True,
         nwp_variables: list[str] | None = None,
+        excarta_variables: list[str] | None = None,
```

Comment:
If we do change it to be the nwp data sources is a list, then I would recommend making `nwp_variables` a list of lists, where each list of variables is in the same order as the data sources.